lib/addons/karpenter/index.ts

```diff
@@ -82,33 +140,36 @@ export class KarpenterAddOn extends HelmAddOn {
             username: 'system:node:{{EC2PrivateDNSName}}'
         });
 
-        // Create Namespace & SA
-        const ns = createNamespace(KARPENTER, cluster);
-        const karpenterPolicyDocument = PolicyDocument.fromJson(KarpenterControllerPolicy);
-        const sa = createServiceAccount(cluster, KARPENTER, KARPENTER, karpenterPolicyDocument);
-        sa.node.addDependency(ns);
+        // Create Namespace
+        const ns = createNamespace(KARPENTER, cluster, true, true);
 
         // Add helm chart
-        setPath(values, "serviceAccount.create", false);
-        setPath(values, "controller.clusterEndpoint", endpoint);
-        setPath(values, "controller.clusterName", name);
-        const karpenterChart = this.addHelmChart(clusterInfo, values, true);
-
-        karpenterChart.node.addDependency(sa);
-
-        // (Optional) default provisioner - defaults to 30 seconds for scale down for
-        // low utilization
-        if (this.options.ProvisionerSpecs){
+        setPath(values, "clusterEndpoint", endpoint);
+        setPath(values, "clusterName", name);
+        values['serviceAccount'] = {
+            create: true,
```

Comment:
If we use standard IRSA account we should be able to state false here and just pass the sa name. We can still use this construct if for whatever reason we want karpenter to create its own service account, but worth a try.  I see nothing special in their helm chart that would require a different approach, but the test will tell: https://github.com/aws/karpenter/blob/main/charts/karpenter/templates/serviceaccount.yaml

Reply:
Addressed, somewhat related to lines 92-116